main.R

```diff
@@ -672,22 +685,19 @@ geco_2022 <- dplyr::mutate(
   )
 )
 
-geco_2022 <- dplyr::ungroup(geco_2022)
-geco_2022 <- dplyr::filter(geco_2022, .data$source == "GECO2022")
-
 
 geco_2022 <- dplyr::mutate(
   geco_2022,
   scenario = dplyr::case_when(
-    grepl("1.5", .data$scenario) ~ "1.5C",
+    grepl(pattern = "1.5", x = .data$scenario) ~ "1.5C",
     grepl("NDC", .data$scenario) ~ "NDC_LTS",
     grepl("Ref", .data$scenario) ~ "Reference",
     TRUE ~ NA_character_
   )
 )
 
 if (any(is.na(unique(geco_2022$scenario)))) {
-  stop("Unique scenario names are not well-defined. Please review!")
+  logger::log_error("`NA` scenario names are not well-defined. Please review!")
```

Comment:
Suggest throwing an error as well as the log message.

Or alternately `log_warn()` and `warning()`

Reply:
I am going to extract the logging/ warning/ erroring-related comments into a new issue, as they are things I would prefer solving after extracting the main body of the function into `pacta.scenario.data.preparation`